src/utilities/aspect-ratio/aspect-ratio.scss

```diff
@@ -0,0 +1,21 @@
+@use 'sass:meta';
+@use '../../compiled/tokens/scss/aspect-ratio';
+@use '../../mixins/media-query';
+
+/// Saving a map of aspect ratio values makes it easier to loop through.
+/// @type Map
+$aspect-ratio-map: meta.module-variables('aspect-ratio');
+
+.u-aspect-ratio {
+  @each $name, $ratio in $aspect-ratio-map {
+    &-#{$name} {
+      @include media-query.breakpoint-classes {
+        --aspect-ratio: #{$ratio};
+        aspect-ratio: var(--aspect-ratio);
+        block-size: 100%;
+        inline-size: 100%;
```

Comment:
Are these properties always necessary?

If so, then this maybe needs to be a component?

Reply:
> Are these properties always necessary?

@tylersticka They seem to both be necessary when using them for a video embed.

> If so, then this maybe needs to be a component? (Edit: Or object?)

Can you elaborate on your thinking process? It's unclear to me why it should be a component or object over a utility class. This will help me in the future when I need to make this decision again. 🙂

Reply:
So according to [Harry Roberts](https://csswizardry.com/2015/03/more-transparent-ui-code-with-namespaces/):

> Utilities are complete [single responsibility](https://csswizardry.com/2012/04/the-single-responsibility-principle-applied-to-css/) rules which have a very specific and targeted task.

[SUIT](https://github.com/suitcss/suit/tree/master/packages/utils#templating) adds a bit more detail to that:

> Each utility class modifies a single trait (a small collection of similar styles).

So if the purpose of these utilities are to apply a specific aspect ratio, I guess the question becomes: Will these additional sizing styles be expected or will they be burdensome?

One possibility would be to use `:is` or `:where` to apply the opinionated convenience styles, so they will be easy to override in case someone wants an element to be `50%` width with a specific aspect ratio?

I guess as I'm writing this, I think this _can_ work as a utility. I just want to make sure that adding these `size` properties won't make the elements harder to resize in contexts where we only want them to fill _some_ of the space?

Reply:
> One possibility would be to use `:is` or `:where` to apply the opinionated convenience styles, so they will be easy to override in case someone wants an element to be `50%` width with a specific aspect ratio?

Ah, thanks for helping me understand the concern! Let me see if I can get it to work. 🔍 👀